### ISSUE TITLE
sale_margin forces giving product_uom to sale line even though normally it is not required

### ISSUE THREAD

**Issue by aisopuro** [odoo/odoo#15580]
#### Impacted versions:
At least 10.0, probably 9.0, not sure about 8.0
#### Steps to reproduce:
1. Start up the Odoo shell to a database that has the `sale` module installed, but not `sale_margin`
1. Create a new sale order.
1. In the shell, create a line for the order and give it only the order and product ids:
    ```python
    self.env['sale.order.line'].create(dict(
        order_id=order_just_created.id,
        product_id=some_product.id
    ))
    ```
1. Now install `sale_margin`
1. Add a line like above using the same parameters

#### Current behavior:
Calling `create` for the second line raises a `KeyError` for the field `product_uom` and the line creation fails.

#### Expected behavior:
Calling `create` with the same parameters as accepted by just the `sale` module should work.

#### Details:
The problem stems from the fact that `sale_margin` assumes that `product_uom` must be in the values given to `create`, since it is a required field: https://github.com/odoo/odoo/blob/10.0/addons/sale_margin/models/sale_order.py#L44

However, the base `create` function in the `sale` module has been updated to handle not being given certain required fields, like `product_uom`: https://github.com/odoo/odoo/blob/10.0/addons/sale/models/sale.py#L678. So `sale_margin` suddenly makes the `create`-method require a field it did not previously require.

**Comment by aisopuro** [comment]
For fixing this, my suggestion would be to move the `purchase_price` calculation in `sale_margin` to after the `super` call. That way we defer deciding which fields are required or not.

I am however not sure if updating `purchase_price` after the initial `create` is somehow problematic. Does @Yenthe666 know of anyone who might give this idea a once-over to make sure I'm not missing some huge pitfall? Greatly appreciated.

**Comment by Yenthe666** [comment]
Hi @aisopuro,

Sorry - it looks like this ping got lost in my messages.
I guess @mart-e could have a look for this one!

Regards,
Yenthe

**Comment by mart-e** [comment]
Should be fixed at b144e09a10e7084cb96c2b420e6b7713834f7866
